src/game/boe.town.hpp

```diff
@@ -21,6 +22,12 @@ void dump_gold(short print_mes);
 void pick_lock(location where,short pc_num);
 void bash_door(location where,short pc_num);
 void erase_specials();
+void erase_hidden_towns(cOutdoors& sector, int quadrant_x, int quadrant_y);
+void erase_completed_specials(cOutdoors& sector);
+
+/*NEW HELPERS, NEED TO GO SOMEWHERE*/
+bool does_location_have_special(cOutdoors& sector, location loc, eTerSpec type);
```

Comment:
Hmm, I like where you're going with this name, but it's a bit long (`location_has_special` maybe?) and there's a slight ambiguity about what "special" means (which may be unavoidable though, since it was already present in the underlying structures).

A logical place for this function is in cOutdoors, or possibly even cArea since it would also make sense calling it on a cTown.